runtime/oti/VMHelpers.hpp

```diff
@@ -2241,21 +2241,28 @@ class VM_VMHelpers
 		/* clear CONCURRENTSCANNING flag bit3:LocalConcurrentScanning /bit4:GlobalConcurrentScanning */
 		ContinuationState oldContinuationState = 0;
 		ContinuationState returnContinuationState = 0;
+		J9VMThread *carrierThread = NULL;
+		bool enteredMonitor = false;
 		do {
 			oldContinuationState = *continuationStatePtr;
+			if (!enteredMonitor) {
```

Comment:
might be an overkill, but if enteredMonitor is true we could assert that  getCarrierThread(oldContinuationState) is still carrierThread

Reply:
we can add assert after move exitConcurrentGCScan() to GCExtensions.cpp

Reply:
ah, we can't assert here because this is *hpp? ok, just don't do it (not even sure that assert after is correct)!?